src/Mod/Sketcher/Gui/SketcherSettings.ui

```diff
@@ -186,6 +186,70 @@ Requires to re-enter edit mode to take effect.</string>
         </property>
        </widget>
       </item>
+      <item>
+       <widget class="Gui::PrefCheckBox" name="checkBoxToggleSnap">
+        <property name="toolTip">
+         <string>Toggle snapping to objects. Snapping is overriding the mouse position to match the position of an object that is within the pick radius.</string>
+        </property>
+        <property name="text">
+         <string>Snap to objects</string>
+        </property>
+        <property name="checked">
+         <bool>true</bool>
+        </property>
+        <property name="prefEntry" stdset="0">
+         <cstring>SnapToObjects</cstring>
+        </property>
+        <property name="prefPath" stdset="0">
+         <cstring>Mod/Sketcher/General</cstring>
+        </property>
+       </widget>
+      </item>
+	  <item>
+       <layout class="QHBoxLayout" name="horizontalLayout">
+        <item>
+         <widget class="QLabel" name="label_1">
+          <property name="minimumSize">
+           <size>
+            <width>182</width>
+            <height>0</height>
+           </size>
+          </property>
+          <property name="text">
+           <string>Angular step for angle snapping (in degrees)</string>
+          </property>
+          <property name="buddy">
+           <cstring>spinBoxSnapAngle</cstring>
+          </property>
+         </widget>
+        </item>
+        <item>
+         <widget class="Gui::PrefDoubleSpinBox" name="spinBoxSnapAngle">
+          <property name="toolTip">
+           <string>Angular step that will be used for tools that have angle snapping (Slot, Line, Arc...) </string>
+          </property>
+          <property name="minimum">
+           <double>0.000000000000000</double>
+          </property>
+          <property name="maximum">
+           <double>360.000000000000000</double>
```

Comment:
I don't see a reason to limit this. While it seems useless to enable above 180, even above 90, it cost nothing to enable up to 360. Capping this to 90 or 180 brings absolutely nothing. It just creates an annoying limitation if someone finds himself in a situation where he wanted to use a greater value.
Even capping at 360 may not be wise, it is perhaps better to leave this uncapped.

Reply:
OK so please do an exercise (will be fast). Draw on a paper how you expect angle snapping to work (from a user perspective) for a value of say 35°. Now compare with the behavior of your code.
You can do same for say 70° and 100°.
You should have good clues what I worry about. ;)

Reply:
I'm not sure if this is what you meant, but there was an issue when trying to snap at more than 180° as the angle was calculated using atan2 is was between -pi and +pi. Meaning that if you tried to snap at 3 times 100° you couldn't as it would go to -100° instead of 300°. Same problem that is solved in the arc tool actually.
I used the same way as it's done in arc.
If you have something else in mind please let me know

Reply:
Actually at the end the problem is to define the expected behavior. As far as I can tell, your code isn't now looking into which direction the mouse is rotating around the reference point, not counting number of turns around as well.

So as a user I would expect the snap to be counted in the -180/+180° range from the reference direction.
For example, with a 70° snapping values, I would expect the possible snapping angle values to be -140;-70;0;+70;+140 and that's all.
But here is only my own opinion.
So we can finally review the PR, you have to specify what is the expected behavior (and eventually got a consensus about that). ;)

Reply:
I thought maybe the best is to act the same way as the arc tool. ie if the user move the mouse clockwise, then he can go the full 360. If he goes counterclockwise same thing.
Well it's what I thought when trying it at least. I'm not sure what others will think about it. It can still be modified later when we have users feedback.

Reply:
Good, now we have a definition of expected behavior. And I saw you (have to) modify the code to match this behavior, so the little exercise hit the target. :)
Now the SnapManager::snapAtAngle is no more deterministic as its computation relies on previous calls.
I guess it's acceptable. Will re-think about it when having another complete review after changes are over.
BTW the  `lastMouseAngle` would better be a static variable scoped inside `snapAtAngle` rather than a member variable. I don't think you will ever need to access it outside `snapAtAngle`.

Reply:
+1 to local static.

Reply:
> +1 to local static.

This said resetting it will be a bit more tricky. :)

Reply:
Hmm indeed it's not reseting if you stop holding shift so it is not working properly. So maybe I need to rollback this change.
Let me know if you think it can work with local static I can't see a cleaner way to do it with local static than it was before.

Reply:
Local static is equivalent from a behavioural point of view. It is cleaner if you do not need this outside the function. If you do need it outside (for resetting if that is necessary). Then it cannot be static.  So you need to determine if you need to use this lastMouseAngle outside the function or not. 

If you only need it inside the function, making it static will remember the last value the variable had when executing the function. It is something like this:

```void myfunc() {
  static double lastMouseAngle
}```
But you need to determine the requirements.

Reply:
There are ways to handle resetting while using a static local, but I can't find one that wouldn't look like a tweak ATM. :)

Reply:
Got it. I'll roll this back then.